cilium/CFP-26135-default-egress-allow.md

```diff
@@ -0,0 +1,198 @@
+# CFP-003: Template
+
+**SIG: SIG-Policy**
+
+**Begin Design Discussion:** 2023-06-12
+
+**Cilium Release:** 1.15
+
+**Authors:** Tamilmani Manoharan <tamanoha@microsoft.com>
+
+## Summary
+
+The current implementation of Cilium lacks a mechanism to restrict egress traffic to specific endpoints without blocking all traffic. When the egress policy is enabled for an endpoint, it effectively goes into a default deny state for egress traffic. This proposal talks about options for adding default egress allow while blocking egress to specific endpoints.
+
+
+## Motivation
+
+In cloud provider environments, it is often necessary to enforce firewall policies to restrict access to certain infrastructure components. Historically, before the emergence of eBPF, firewall rules were enforced using IPTables. Cilium primarily utilizes eBPF for policy enforcement. In eBPF host routing mode, Cilium bypasses IPtable rules and rules can be enforced only via eBPF. Also in certain scenarios (eg: L7), cilium prepends some IPTable rules by default which conflicts with cloud provider specific rules. Disabling the "prepend-iptables-chains" option is also not a viable solution as disabling prepend means Cilium can no longer guarantee that its policies are enforced.
+
+This proposal aims to bring the management of cloud provider specific policies within Cilium. The primary goal of this proposal is to integrate the cloud provider-specific policies into the Cilium framework. By doing so, we aim to address potential conflicts and other concerns associated with managing these policies across multiple components.
+
+## Goals
+
+- Block egress traffic to certain endpoints without blocking all
+- Behavior of existing network policy or cilium network policy should not be affected
+
+## Non Goals
+ - Ingress Restriction without blocking all
+ - Applying this policy for specific endpoints
+ - Complete Network Policy support with all features
+
+## Proposal
+
+### Overview
+The proposal will talk about preferred option and also about other options that aims to solve the problem by allowing cloud provider to specify the intent of blocking egress to certain endpoints without blocking all. These deny rules can be plumbed either via iptables or bpf policy rules. IPtables rules would not work with ebpf host routing mode. This proposal considers to solve this via bpf policy rules as it will work for both legacy host routing and ebpf host routing modes. It will also ensure that the behavior of existing network policies or Cilium network policies is not regressed.
```

Comment:
nit, there's a lot of words here that are not giving an overview of the solution. This can make it harder for other contributors to understand the proposal.
```suggestion
Allow cloud providers to block egress traffic to certain endpoints without blocking all. This proposal considers to solve this via bpf policy rules as it will work for both legacy host routing and ebpf host routing modes.
```